server/src/main/java/org/candlepin/bind/HandleCertificatesOp.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2009 - 2012 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.bind;
+
+import org.candlepin.controller.EntitlementCertificateGenerator;
+import org.candlepin.model.Entitlement;
+import org.candlepin.model.EntitlementCertificate;
+import org.candlepin.model.EntitlementCertificateCurator;
+import org.candlepin.model.EntitlementCurator;
+import org.candlepin.model.Pool;
+import org.candlepin.model.PoolQuantity;
+import org.candlepin.model.Product;
+
+import com.google.inject.Inject;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This bind operation is responsible to create certificates and regenerate
+ * relevant certificates during a bind.
+ *
+ * TODO: either fetch ents to regenerate in preProcess and then mark them dirty in execute,
+ * otherwise, write a query to directly update ents without fetching them, all in execute method.
```

Comment:
@Ceiu , thats an upcoming PR. I will re base #1542 and re-work on it after this is merged.